both/collections/competition.js

```diff
@@ -46,13 +49,19 @@ Competitions.attachSchema({
     max: 23.5*60,
     defaultValue: 23.5*60,
     custom: function() {
-      // require the competition calendar end time to be after the start time
-      var calendarStartMinutes = this.field("calendarStartMinutes").value;
-      var calendarEndMinutes = this.value;
-      if(calendarEndMinutes <= calendarStartMinutes) {
+      var obj = validationObject(this, ['calendarStartMinutes', 'calendarEndMinutes']);
+
+      if(obj.calendarEndMinutes <= obj.calendarStartMinutes) {
         return "calendarEndIsNotBeforeStart";
       }
-      return null;
+
+      var error = null;
+      ScheduleEvents.find({competitionId: this.docId}).forEach(function(event) {
+        if(event.endMinutes() > obj.calendarEndMinutes) {
+          error = "laterExistingEvents";
```

Comment:
Yeah, that's the illusion I was also under for longer than I'm happy about.

But that only returns from the `forEach()` function, and there isn't even a way to get to the returned value.

As I rubber duck this, I realize there *are* other ways. I like this one:

      var events = ScheduleEvents.find({competitionId: this.docId}).fetch();
      var found = _.find(events, function(event){ return event.endMinutes() > obj.calendarEndMinutes; });
      return found ? "laterExistingEvents" : null;